### PR TITLE
 Improve variable names in map_can_construct_with_clear_at() and use big Z

### PR DESCRIPTION
**Pull request by tupaschoal**
This is a followup to @duncanspumpkin on https://github.com/OpenRCT2/OpenRCT2/pull/10462#discussion_r361832868

### CODE REVIEW

File: src/openrct2/world/Map.cpp [hunk 1]
```diff
@@ -1312,14 +1312,11 @@ bool map_can_construct_with_clear_at(
     loc_68B9B7:
         if (gParkFlags & PARK_FLAGS_FORBID_HIGH_CONSTRUCTION)
         {
-            al = pos.clearanceZ / 8 - tileElement->base_height;
-            if (al >= 0)
+            northZ = pos.clearanceZ - tileElement->GetBaseZ();
+            if (northZ >= 0 && northZ > (18 * 8))
```
Comment (Gymnasiast) [inline src/openrct2/world/Map.cpp]:
```suggestion
            if (northZ >= 0 && northZ > (18 * COORDS_Z_STEP))
```

File: src/openrct2/world/Map.cpp [hunk 2]
```diff
@@ -1340,43 +1337,43 @@ bool map_can_construct_with_clear_at(
...
-                ch = al;
+                northZ = tileElement->GetBaseZ();
+                eastZ = northZ;
+                southZ = northZ;
+                westZ = northZ;
                 slope = tileElement->AsSurface()->GetSlope();
                 if (slope & TILE_ELEMENT_SLOPE_N_CORNER_UP)
                 {
-                    al += 2;
+                    northZ += (2 * 8);
```
Comment (Gymnasiast) [inline src/openrct2/world/Map.cpp]: Also use COORDS_Z_STEP on all these.

File: src/openrct2/world/Map.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/world/Map.cpp]: Also this instance of variable al should be renamed to `heightFromGround`. I was making that change anyway in the #10501

Comment (tupaschoal) [reply src/openrct2/world/Map.cpp]: Ok, did both.

**Pull request merged**